### DISCUSSION TITLE
Is it possible to use bwapi on MacOS ?

### DISCUSSION THREAD

**Discussion by Virgile-Foussereau** [bwapi/bwapi#897] [Q&A] [answered]
Hello,

Is it possible to use bwapi on Mac ?

Thank you

**Comment by dgant** [comment, marked as answer]
StarCraft and BWAPI only run in Windows, but you can develop on other operating systems with some extra steps. You can run games or compile C++ bots using WINE, Docker, or virtual machines. [SC-Docker](https://github.com/basil-ladder/sc-docker) runs BWAPI games on multiple platforms.

[StardustDevEnvironment](https://github.com/bmnielsen/StardustDevEnvironment) is for writing bots in Linux or MacOS. It uses [OpenBW](https://github.com/OpenBW/openbw) which is an alternative version of StarCraft and BWAPI which is cross-platform, but not immediately compatible with bots compiled for StarCraft 1.16.1. StardustDevEnvironment lets you test locally on OpenBW and then compile a Windows binary for competitions like the SSCAI Tournament.

(From the [SSCAIT Tutorial](https://sscaitournament.com/index.php?action=tutorial))

I believe you can also develop in Boot Camp.
